### PR TITLE
Don't initialise dict for same user multiple times 

### PR DESCRIPTION
**Pull request by vansika**
user data dictionary in candidate_sets.py for same user was being initialized multiple times leading to incomplete information in candidate HTML. Have introduced a simple if condition and added a basic test.

### CODE REVIEW

File: listenbrainz_spark/recommendations/tests/test_candidate.py [hunk 1]
```diff
@@ -259,3 +259,193 @@ def test_save_candidate_sets(self):
...
+                mb_recording_mbid='yyy',
+                msb_artist_credit_name_matchable='lessthanjake',
+                msb_recording_name_matchable='lalal',
+                recording_id=2,
+                user_name='vansika',
+            ),
+            schema=None
+        ))
+
+        recieved_user_data = candidate_sets.get_candidate_html_data(similar_artist_candidate_set_df_html,
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/tests/test_candidate.py:402]: received

Comment (vansika) [reply listenbrainz_spark/recommendations/tests/test_candidate.py:402]: Ughhhh.

Comment (vansika) [reply listenbrainz_spark/recommendations/tests/test_candidate.py:402]: I swear I know the spelling. You believe me right?

File: listenbrainz_spark/recommendations/tests/test_candidate.py [hunk 2]
```diff
@@ -259,3 +259,193 @@ def test_save_candidate_sets(self):
 
         similar_artist_exist = utils.path_exists(path.SIMILAR_ARTIST_CANDIDATE_SET)
         self.assertTrue(top_artist_exist)
+
+    def test_get_candidate_html_data(self):
+        df = utils.create_dataframe(
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/tests/test_candidate.py:264]: The re-use of df is confusing there.

File: listenbrainz_spark/recommendations/tests/test_candidate.py (hunk 1 above)
Comment (mayhem) [reply listenbrainz_spark/recommendations/tests/test_candidate.py:402]: Doesn't really matter does it? 😆

**Pull request merged**